python/ray/actor.py

```diff
@@ -367,13 +392,33 @@ def invocation(args, kwargs):
                     _generator_backpressure_num_objects
                 ),
                 enable_task_events=enable_task_events,
+                tensor_transport=tensor_transport,
             )
 
         # Apply the decorator if there is one.
         if self._decorator is not None:
             invocation = self._decorator(invocation)
 
-        return invocation(args, kwargs)
+        obj_ref = invocation(args, kwargs)
+        if tensor_transport != "OBJECT_STORE":
+            # print(f"_remote tensor_transport: {tensor_transport}")
+            assert num_returns == 1
```

Comment:
Raise a ValueError or NotImplementedError here instead (asserts should only be used for system-level errors).

Reply:
Raise a ValueError: https://github.com/ray-project/ray/pull/52938/commits/de7ce021f9c4db0bee0c3670521bd51ec796b54f